Test failure with test_blackbody_scipy on 32-bit

I'm seeing the following failure on the new 32-bit CircleCI set-up (#7126):

```python
_____________________________ test_blackbody_scipy _____________________________
[gw2] linux -- Python 3.6.4 /opt/python/cp36-cp36m/bin/python

    @pytest.mark.skipif('not HAS_SCIPY')
    def test_blackbody_scipy():
        """Test Planck function.
    
        .. note:: Needs ``scipy`` to work.
    
        """
        flux_unit = u.Watt / (u.m ** 2 * u.um)
        wave = np.logspace(0, 8, 100000) * u.AA
        temp = 100. * u.K
        with np.errstate(all='ignore'):
            bb_nu = blackbody_nu(wave, temp) * u.sr
        flux = bb_nu.to(flux_unit, u.spectral_density(wave)) / u.sr
    
        lum = wave.to(u.um)
        intflux = integrate.trapz(flux.value, x=lum.value)
        ans = const.sigma_sb * temp ** 4 / np.pi
>       np.testing.assert_allclose(intflux, ans.value, rtol=0.01)  # 1% accuracy

astropy/modeling/tests/test_blackbody.py:73: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

x_id = True, y_id = False, hasval = 'nan'

    def chk_same_position(x_id, y_id, hasval='nan'):
        """Handling nan/inf: check that x and y have the nan/inf at the same
            locations."""
        try:
            assert_array_equal(x_id, y_id)
        except AssertionError:
            msg = build_err_msg([x, y],
                                err_msg + '\nx and y %s location mismatch:'
                                % (hasval), verbose=verbose, header=header,
                                names=('x', 'y'), precision=precision)
>           raise AssertionError(msg)
E           AssertionError: 
E           Not equal to tolerance rtol=0.01, atol=0
E           
E           x and y nan location mismatch:
E            x: array(nan)
E            y: array(1.804934)

/opt/python/cp36-cp36m/lib/python3.6/site-packages/numpy/testing/nose_tools/utils.py:706: AssertionError
```

@pllim - do you fancy investigating this? If you don't have time, I can take a look.

--- a/astropy/modeling/blackbody.py
+++ b/astropy/modeling/blackbody.py
@@ -113,7 +113,7 @@ FLAM = u.erg / (u.cm**2 * u.s * u.AA)
 # See https://github.com/astropy/astropy/issues/4171
 with warnings.catch_warnings():
     warnings.simplefilter('ignore', RuntimeWarning)
-    _has_buggy_expm1 = np.isnan(np.expm1(1000))
+    _has_buggy_expm1 = np.isnan(np.expm1(1000)) or np.isnan(np.expm1(1e10))
 
 
 class BlackBody1D(Fittable1DModel):

